src/test/integration/things-test.ts

```diff
@@ -1021,6 +1035,70 @@ describe('things/', function () {
     expect(res.body[0].a).toHaveProperty('timestamp');
   });
 
+  it('should be able to subscribe to events using EventSource', async () => {
+    await addDevice(EVENT_THING);
+    if (!httpServer.address()) {
+      httpServer.listen();
+      await e2p(httpServer, 'listening');
+    }
+    const addr = <AddressInfo>httpServer.address()!;
+
+    // Test event subscription
+    let eventSourceURL =
+      `http://127.0.0.1:${addr.port}${Constants.THINGS_PATH}/` +
+      `${EVENT_THING.id}/events/overheated?jwt=${jwt}`;
+    const eventSource = new EventSource(eventSourceURL) as EventTarget & EventSource;
+    await e2p(eventSource, 'open');
+    const overheatedEvent = new Event('overheated', 101, EVENT_THING.id);
+    const [, event] = await Promise.all([
+      Events.add(overheatedEvent),
+      e2p(eventSource, 'overheated'),
+    ]);
+    expect(event.type).toEqual('overheated');
+    expect(JSON.parse(event.data)).toEqual(101);
+    eventSource.close();
+
+    // Test events subscription
+    eventSourceURL =
+      `http://127.0.0.1:${addr.port}${Constants.THINGS_PATH}/` +
+      `${EVENT_THING.id}/events?jwt=${jwt}`;
+    const eventsSource = new EventSource(eventSourceURL) as EventTarget & EventSource;
+    await e2p(eventsSource, 'open');
+    const overheatedEvent2 = new Event('overheated', 101, EVENT_THING.id);
+    const [, event2] = await Promise.all([
+      Events.add(overheatedEvent2),
+      e2p(eventsSource, 'overheated'),
+    ]);
+    expect(event2.type).toEqual('overheated');
+    expect(JSON.parse(event2.data)).toEqual(101);
+    eventsSource.close();
+
+    // Test non-existent thing errors
+    eventSourceURL =
+      `http://127.0.0.1:${addr.port}${Constants.THINGS_PATH}` +
+      `/non-existent-thing/events/overheated?jwt=${jwt}`;
+    const thinglessEventSource = new EventSource(eventSourceURL) as EventTarget & EventSource;
+    thinglessEventSource.onerror = jest.fn();
+    thinglessEventSource.onopen = jest.fn();
+    await e2p(thinglessEventSource, 'error');
+    expect(thinglessEventSource.onopen).not.toBeCalled();
+    expect(thinglessEventSource.onerror).toBeCalled();
+
+    // Test non-existent event errors
+    eventSourceURL =
+      `http://127.0.0.1:${addr.port}${Constants.THINGS_PATH}` +
+      `${EVENT_THING.id}/events/non-existentevent?jwt=${jwt}`;
+    const eventlessEventSource = new EventSource(eventSourceURL) as EventTarget & EventSource;
+    eventlessEventSource.onerror = jest.fn();
+    eventlessEventSource.onopen = jest.fn();
+    await e2p(eventlessEventSource, 'error');
+    expect(eventlessEventSource.onopen).not.toBeCalled();
+    expect(eventlessEventSource.onerror).toBeCalled();
+
+    // Clean up
+    httpServer.close();
```

Comment:
As above server lifecycle is somewhat handled in `../common` :
```ts
afterAll(async () => {
  Logs.close();
  await addonManager.unloadAddons();

  if (servers.https) {
    servers.https.close();
    await e2p(servers.https, 'close');
  }

  servers.http.close();
  await e2p(servers.http, 'close');

  removeTestManifest();
});
```
Do we really need this?